source/ui/CLS/CLSSIS3820ScalerControlsView.cpp

```diff
@@ -0,0 +1,246 @@
+	timeLayout->addWidget(time_);
+
+	QHBoxLayout *scansPerBufferLayout = new QHBoxLayout;
+	scansPerBufferLayout->addWidget(new QLabel("Scans per Buffer:"), 0, Qt::AlignRight);
+	scansPerBufferLayout->addWidget(scansPerBuffer_);
+
+	QHBoxLayout *totalScansLayout = new QHBoxLayout;
+	totalScansLayout->addWidget(new QLabel("Total Scans"), 0, Qt::AlignRight);
+	totalScansLayout->addWidget(totalScans_);
+
+	QVBoxLayout *spinBoxLayout = new QVBoxLayout;
+	spinBoxLayout->addLayout(timeLayout);
+	spinBoxLayout->addLayout(scansPerBufferLayout);
+	spinBoxLayout->addLayout(totalScansLayout);
+
+	QHBoxLayout *layout = new QHBoxLayout;
+	layout->addLayout(statusAndModeLayout);
+	layout->addLayout(spinBoxLayout);
+
+	setLayout(layout);
+
+	// Make connections.
+
+	connect(startButton_, SIGNAL(clicked()), this, SLOT(startScanning()));
+	connect(stopButton_, SIGNAL(clicked()), this, SLOT(stopScanning()));
+	connect(modeChoice_, SIGNAL(currentIndexChanged(int)), this, SLOT(setContinuous()));
+	connect(time_, SIGNAL(editingFinished()), this, SLOT(setDwellTime()));
+	connect(scansPerBuffer_, SIGNAL(editingFinished()), this, SLOT(setScansPerBuffer()));
+	connect(totalScans_, SIGNAL(editingFinished()), this, SLOT(setTotalScans()));
+
+	// Current settings.
+
+	setScaler(scaler);
+}
+
+CLSSIS3820ScalerControlsView::~CLSSIS3820ScalerControlsView()
+{
+
+}
+
+void CLSSIS3820ScalerControlsView::setScaler(CLSSIS3820Scaler *newScaler)
+{
+	if (scaler_ != newScaler) {
+
+		if (scaler_) {
+			disconnect( scaler_, 0, this, 0 );
+		}
+
+		scaler_ = newScaler;
+
+		if (scaler_) {
+			connect(scaler_, SIGNAL(scanningChanged(bool)), this, SLOT(onScanningChanged()) );
+			connect(scaler_, SIGNAL(continuousChanged(bool)), this, SLOT(onContinuousChanged()));
+			connect(scaler_, SIGNAL(dwellTimeChanged(double)), this, SLOT(onDwellTimeChanged()));
+			connect(scaler_, SIGNAL(scansPerBufferChanged(int)), this, SLOT(onScansPerBufferChanged()));
+			connect(scaler_, SIGNAL(totalScansChanged(int)), this, SLOT(onTotalScansChanged()));
+		}
+
+		onScanningChanged();
+		onContinuousChanged();
+		onDwellTimeChanged();
+		onScansPerBufferChanged();
+		onTotalScansChanged();
+	}
+}
+
+void CLSSIS3820ScalerControlsView::startScanning()
+{
+	if (scaler_)
+		scaler_->setScanning(true);
+}
+
+void CLSSIS3820ScalerControlsView::stopScanning()
+{
+	if (scaler_)
+		scaler_->setScanning(false);
+}
+
+void CLSSIS3820ScalerControlsView::setContinuous()
+{
+	if (scaler_) {
+		scaler_->blockSignals(true);
+
+		int index = modeChoice_->currentIndex();
+
+		if (index == 1 && !scaler_->isContinuous())
+			scaler_->setContinuous(true);
+
+		else if (index == 0 && scaler_->isContinuous())
+			scaler_->setContinuous(false);
+
+		scaler_->blockSignals(false);
+	}
+}
+
+void CLSSIS3820ScalerControlsView::setDwellTime()
+{
+	if (scaler_) {
+		scaler_->blockSignals(true);
+
+		double secondsDwell = (double)time_->value() / MILLISECONDS_PER_SECOND;
```

Comment:
You could just make this 1000 since I don't think there is any disabiguation required for millisecond to second conversions.